sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -618,6 +650,47 @@ <T> CreationState needToCreate(T byName, T byPort, String name) {
         }
     }
 
+    @Override
+    public ApplicationGatewayImpl withExistingWebApplicationFirewallPolicy(WebApplicationFirewallPolicy wafPolicy) {
+        ensureWafV2();
+        if (wafPolicy != null) {
+            return withExistingWebApplicationFirewallPolicy(wafPolicy.id());
+        }
+        return this;
+    }
+
+    @Override
+    public ApplicationGatewayImpl withExistingWebApplicationFirewallPolicy(String resourceId) {
+        ensureWafV2();
+        if (resourceId != null) {
+            this.innerModel().withFirewallPolicy(new SubResource().withId(resourceId));
+        }
+        return this;
+    }
+
+    @Override
+    public ApplicationGatewayImpl withNewWebApplicationFirewallPolicy(boolean enable, WebApplicationFirewallMode mode) {
+        ensureWafV2();
+        WebApplicationFirewallPolicy.DefinitionStages.WithCreate wafPolicyCreatable = this.manager().webApplicationFirewallPolicies()
+            .define(this.manager().resourceManager().internalContext().randomResourceName("wafpolicy", 14))
```

Comment:
match Portal default naming prefix, if possible.

Reply:
Portal requires the name of the policy when creating Application Gateway. 

PowerShell in their example uses `WafPol` and `wafpolicy` as the name prefix.
https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/tutorial-restrict-web-traffic-powershell#create-the-application-gateway-with-the-waf

Any suggestion?

Reply:
I don't have good suggestion, Maybe we just use the current one. Or maybe try one's luck on CLI (if they create it automatically).

Reply:
CLI doesn't create automatically either. They use `waf-pol` in their sample:
https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/tutorial-restrict-web-traffic-cli#create-an-application-gateway-with-a-waf-policy

Maybe we just use the current...